contracts/implementations/nodes/AftermarketDevice.sol

```diff
@@ -258,24 +259,45 @@ contract AftermarketDevice is
     ) external onlyRole(Roles.CLAIM_AD_ROLE) {
         AftermarketDeviceStorage.Storage storage ads = AftermarketDeviceStorage
             .getStorage();
-        bytes32 message = keccak256(
-            abi.encode(CLAIM_TYPEHASH, aftermarketDeviceNode, owner)
-        );
         address aftermarketDeviceAddress = ads.nodeIdToDeviceAddress[
             aftermarketDeviceNode
         ];
+        bytes32 messageOwner = keccak256(
+            abi.encode(
+                CLAIM_TYPEHASH,
+                aftermarketDeviceNode,
+                owner,
+                NoncesInternal._useNonce(CLAIM_TYPEHASH, owner)
+            )
+        );
+        bytes32 messageDevice = keccak256(
+            abi.encode(
+                CLAIM_TYPEHASH,
+                aftermarketDeviceNode,
+                owner,
+                NoncesInternal._useNonce(
+                    CLAIM_TYPEHASH,
+                    aftermarketDeviceAddress
```

Comment:
My bad on this. I think we need to do one nonce, the user’s, for now. As long as the message contains both the user’s address and the user’s nonce, I think we are okay.